BookOfHabits/BookOfHabitsMicroservice.Application.Service/CardsApplicationService.cs

```diff
@@ -0,0 +1,104 @@
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(id, nameof(Card)));
+            if (await cardRepository.DeleteAsync(card) is false)
+                throw new BadRequestException(FormatBadRequestErrorMessage(id, nameof(Card)));
+        }
+
+        public async Task<IEnumerable<CardModel>> GetAllCardsAsync(CancellationToken token = default)
+        {
+            var cards = (await cardRepository.GetAllAsync(filter: x => x.IsPublic,
+                                                          asNoTracking: true, 
+                                                          cancellationToken: token));
+            return cards.Select(mapper.Map<CardModel>);
+        }
+
+        public async Task<CardModel?> GetCardByIdAsync(Guid id, CancellationToken token = default)
+        {
+            Card card = await cardRepository.GetByIdAsync(x => x.Id.Equals(id),
+                                                                includes: $"{nameof(Card.TemplateValues)}",
+                                                                asNoTracking: true,
+                                                                cancellationToken: token)
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(id, nameof(Card)));
+            return mapper.Map<CardModel>(card);
+        }
+
+        public async Task UpdateCard(UpdateCardModel cardInfo, CancellationToken token = default)
+        {
+            var card = await cardRepository.GetByIdAsync(x => x.Id.Equals(cardInfo.Id), cancellationToken: token)
+                 ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(cardInfo.Id, nameof(Card)));
+            if (cardInfo.Name is not null)
+                card.SetName(cardInfo.Name);
+            if (cardInfo.Description is not null)
+                card.SetDescription(cardInfo.Description);
+            if (cardInfo.Image is not null)
+                card.SetImage(cardInfo.Image);
+            if (cardInfo.TitleCheckElements is not null)
+                card.SetTitlesCheck(cardInfo.TitleCheckElements);
+            card.SetOptions(cardInfo.Options);
+            await cardRepository.UpdateAsync(entity: card, token);
+        }
+
+        public async Task UpdateTemplateValues(Guid cardId, UpdateTemplateValuesModel tempateValuesInfo, CancellationToken token = default)
+        {
+            var card = await cardRepository.GetByIdAsync(x => x.Id.Equals(cardId), includes: nameof(Card.TemplateValues), cancellationToken: token)
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(cardId, nameof(Card)));
+            var template = await templateValuesRepository.GetByIdAsync(x => x.Id.Equals(card.TemplateValues.Id), cancellationToken: token)
+                ?? throw new NotFoundException(FormatFullNotFoundErrorMessage(card.TemplateValues.Id, nameof(TemplateValues)));
+            if (tempateValuesInfo.Status is not null)
```

Comment:
I have no clue about logic, but the multitude of 'if' it seems like 'smell code' and should be refactored